newscoop/library/Newscoop/Services/ThemesService.php

```diff
@@ -91,16 +105,51 @@ public function getThemePath()
                 $this->cacheService->save($cacheKeyWebOutput, $webOutput);
             }
 
-            $outSetIssues = $this->findByIssueAndOutput($issue->getId(), $webOutput);
-            if (!is_null($outSetIssues)) {
-                $themePath = $outSetIssues->getThemePath()->getPath();
+            if ($frontpage) {
+                $outSetPublication = $this->findByPublicationAndLanguageAndOuput($publication, $languageId, $webOutput);
+                if (!is_null($outSetPublication)) {
+                    $themePath = $outSetPublication->getThemePath()->getPath();
+                }
+            }
+            if (!$frontpage || ($frontpage && $themePath === null)) {
+                $outSetIssues = $this->findByIssueAndOutput($issue->getId(), $webOutput);
+                if (!is_null($outSetIssues)) {
+                    $themePath = $outSetIssues->getThemePath()->getPath();
+                }
             }
 
             $this->cacheService->save($cacheKeyThemePath, $themePath);
+        }
 
+        return $themePath;
+    }
+
+    /**
+     * Returns whether the url is the home page or not. Support for
+     * language homepages (/cz, /en, etc.) is also built in via a parameters.
+     *
+     * @param  boolean $withLanguageHomepage Whether to include language homepages
+     *                                       in the resulsts.
+     *
+     * @return boolean
+     */
+    public function isHomepage($path, $withLanguageHomepage = false)
+    {
+        $matches = array();
+        if ($path === '/') {
+            return true;
+        } elseif ($withLanguageHomepage && preg_match('@^/[a-z]{2}(?:_[A-Z]{2})?$@', $path, $matches)) {
```

Comment:
I know i made the regex, but it should be @^/[a-z]{2}(?:\-[A-Z]{2})?$@ since we store language as en-US in our db. Or maybe we can check for en_US and en-US?